src/lib/operators/table_scan/abstract_single_column_table_scan_impl.cpp

```diff
@@ -0,0 +1,77 @@
+#include "abstract_single_column_table_scan_impl.hpp"
+
+#include <memory>
+#include <unordered_map>
+#include <utility>
+
+#include "resolve_type.hpp"
+#include "storage/chunk.hpp"
+#include "storage/dictionary_segment.hpp"
+#include "storage/reference_segment.hpp"
+#include "storage/split_pos_list_by_chunk_id.hpp"
+#include "storage/table.hpp"
+#include "storage/value_segment.hpp"
+
+namespace opossum {
+
+AbstractSingleColumnTableScanImpl::AbstractSingleColumnTableScanImpl(const std::shared_ptr<const Table>& in_table,
+                                                                     const ColumnID column_id,
+                                                                     const PredicateCondition predicate_condition)
+    : _in_table(in_table), _column_id(column_id), _predicate_condition(predicate_condition) {}
+
+std::shared_ptr<PosList> AbstractSingleColumnTableScanImpl::scan_chunk(const ChunkID chunk_id) const {
+  const auto& chunk = _in_table->get_chunk(chunk_id);
+  const auto& segment = chunk->get_segment(_column_id);
+
+  auto matches = std::make_shared<PosList>();
```

Comment:
what do you think of all these Scan functions **returning** a PosList (and an instance, not a shared_ptr), so we have a nice function input/output separation instead of having the output of the function as the third of four parameters?

Reply:
Or should we at least (since my proposal above would probably be slow in the case of a scan on a ReferenceSegment that references multiple Chunks) make some of these class members instead of passing them around?

Reply:
We could just override the members at the beginning of `scan_chunk` - passing less parameters around would be nice, but I won't press for this.

Reply:
Wouldn't that break multithreading?

Reply:
Ah, that's why. Doc that, pls